Atomic/discovery.py

```diff
@@ -52,7 +53,10 @@ def assemble_fqdn(self, include_tag=True, registry=None):
         fqdn = fqdn if not self.repo else "{}/{}".format(fqdn, self.repo)
         fqdn += "/{}".format(self.image)
         if include_tag:
-            fqdn += ":{}".format(self.tag)
+            if self.tag:
```

Comment:
I think the right place for this would be in util.Decompose, I will add something there.